File: app/src/main/java/com/coinninja/android/helpers/Resources.java
package com.coinninja.android.helpers;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.TypedValue;

public class Resources {
    public static String getString(Context context, int resourceId) {
        return context.getResources().getString(resourceId);
    }

    public static String getString(Context context, int resourceId, String... formats) {
        return context.getResources().getString(resourceId, (Object[]) formats);
    }

    public static Drawable getDrawable(Context context, int resourceId) {
        return context.getResources().getDrawable(resourceId);
    }

    public static int getColor(Context context, int resourceId) {
        return context.getResources().getColor(resourceId);
    }

    public static float scaleValue(Context context, int unit, float value) {
        return TypedValue.applyDimension(unit, value, context.getResources().getDisplayMetrics());
    }
}

File: app/src/main/java/com/coinninja/android/helpers/Resources.kt
package com.coinninja.android.helpers

import android.content.Context
import android.graphics.drawable.Drawable
import android.util.TypedValue
import androidx.appcompat.content.res.AppCompatResources

object Resources {
    fun getString(context: Context, resourceId: Int): String {
        return context.resources.getString(resourceId)
    }

    fun getString(context: Context, resourceId: Int, vararg formats: String): String {
        return context.resources.getString(resourceId, *formats as Array<Any>)
    }

    fun getDrawable(context: Context, resourceId: Int): Drawable? {
        val drawable = AppCompatResources.getDrawable(context, resourceId)
        drawable?.setBounds(0, 0, drawable.intrinsicWidth, drawable.intrinsicHeight)
        return drawable
    }

    fun getColor(context: Context, resourceId: Int): Int {
        return context.resources.getColor(resourceId)
    }

    fun scaleValue(context: Context, unit: Int, value: Float): Float {
        return TypedValue.applyDimension(unit, value, context.resources.displayMetrics)
    }
}
